sentry-log4j2/src/main/java/io/sentry/log4j2/SentryAppender.java

```diff
@@ -0,0 +1,220 @@
+package io.sentry.log4j2;
+
+import io.sentry.core.Breadcrumb;
+import io.sentry.core.DateUtils;
+import io.sentry.core.Sentry;
+import io.sentry.core.SentryEvent;
+import io.sentry.core.SentryLevel;
+import io.sentry.core.SentryOptions;
+import io.sentry.core.protocol.Message;
+import io.sentry.core.protocol.SdkVersion;
+import io.sentry.core.transport.ITransport;
+import io.sentry.core.util.CollectionUtils;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.Date;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.Optional;
+import java.util.stream.Collectors;
+import org.apache.logging.log4j.Level;
+import org.apache.logging.log4j.core.Filter;
+import org.apache.logging.log4j.core.LogEvent;
+import org.apache.logging.log4j.core.appender.AbstractAppender;
+import org.apache.logging.log4j.core.config.plugins.Plugin;
+import org.apache.logging.log4j.core.config.plugins.PluginAttribute;
+import org.apache.logging.log4j.core.config.plugins.PluginElement;
+import org.apache.logging.log4j.core.config.plugins.PluginFactory;
+import org.apache.logging.log4j.core.impl.ThrowableProxy;
+import org.jetbrains.annotations.ApiStatus;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+
+/** Appender for Log4j2 in charge of sending the logged events to a Sentry server. */
+@Plugin(name = "Sentry", category = "Core", elementType = "appender", printObject = true)
+public final class SentryAppender extends AbstractAppender {
+  private @Nullable String dsn;
+  private @Nullable ITransport transport;
+  private @NotNull Level minimumBreadcrumbLevel = Level.INFO;
+  private @NotNull Level minimumEventLevel = Level.ERROR;
```

Comment:
does it make sense to add this either to options or to the Level class as static default fields and we just read the default values from them? cus I see these 2 values repeated in all the integrations, like:

```
SentryOptions.DEFAULT_BREADCRUMB_LEVEL
SentryOptions.DEFAULT_EVENT_LEVEL
```

or something like that.

Reply:
It depends if we want to pollute `sentry-core` with integration specific code. At this stage core is not aware of the whole mechanism of attaching breadcrumbs to events in logging integrations.

Reply:
I'd be fine with that, to not DRY, of course, it depends on a case by case, but this duplication is already in 4 integrations.

It'd be something similar, defining default values to be reused elsewhere.

Reply:
Ok understood. This would mean that we need to add conversion from SentryLevel to logging-framework-level - as these levels in appenders are are from log4j2/Logback packages.

Reply:
yeah the converters for sure, it's just to reuse the default levels, nothing more.

Reply:
IMHO I also rather to keep them out since core has nothing to do with min breadcrumb level of a logger integration

DRY doesn't mean avoid any and all "duplication" by breaking cohesion